ansible/playbooks/roles/upgrade/defaults/main.yml

```
@@ -19,3 +19,7 @@ opendistro_for_elasticsearch:
       filename: demo2epiphany-certs-migration-root-CAs.pem
 
   upgrade_state_file_path: /etc/elasticsearch/epicli-upgrade-started.state
+
+kubernetes:
+  upgrade_master_state_file_path: /var/lib/epiphany/upgrade/state/kubernetes-master-{{ ver }}.uncompleted
+  upgrade_node_state_file_path: /var/lib/epiphany/upgrade/state/kubernetes-node-{{ ver }}.uncompleted
```

```suggestion
kubernetes:
  upgrade_state_file_path: /var/lib/epiphany/upgrade/state/kubernetes-{{ ver }}.uncompleted
```
kubernetes_master and kubernetes_node roles are exclusive so how about common file?

---

Condition for the block **Upgrade nodes** [here](https://github.com/epiphany-platform/epiphany/pull/2766/files#diff-b642cd0642ad945c5f7f283861c694a6c5fb4218ed56c80eabf9df27fecd615bR81-R83) doesn't check if we are operating on master or node inventory_hostname, therefore it always enters block, creates flag file, skips upgrade, and then removes flag file again -> so even if the flag was present because of the master failure, it was removed inside upgrade **node block**. 

I will add all of your suggested changes, along with the additional change in [this condition](https://github.com/epiphany-platform/epiphany/pull/2766/files#diff-b642cd0642ad945c5f7f283861c694a6c5fb4218ed56c80eabf9df27fecd615bR81-R83), where we will check based on inventory hostname if we have to enter **node upgrade** block.